pandas/tests/arrays/string_/test_string_arrow.py

```diff
@@ -83,11 +83,14 @@ def test_constructor_not_string_type_value_dictionary_raises(chunked):
         ArrowStringArray(arr)
 
 
+@pytest.mark.xfail(
+    reason="dict conversion does not seem to be implemented for large string in arrow"
+)
 @pytest.mark.parametrize("chunked", [True, False])
 def test_constructor_valid_string_type_value_dictionary(chunked):
     pa = pytest.importorskip("pyarrow")
 
-    arr = pa.array(["1", "2", "3"], pa.dictionary(pa.int32(), pa.utf8()))
+    arr = pa.array(["1", "2", "3"], pa.dictionary(pa.int32(), pa.large_string()))
```

Comment:
```suggestion
    arr = pa.array(["1", "2", "3"], pa.large_string()).dictionary_encode()
```

(it's only the python->arrow converter that doesn't seem to implement this, but creating a dictionary array with large string in pyarrow itself is certainly supported)

Reply:
Additionally, it looks a bit strange that we actually allow creating a string column backed by a dictionary array. It would be nice that long-term we support this, but right now many operations will just fail (eg all string compute functions from pyarrow will fail on a dictionary[string] type). 

I think for fixing https://github.com/pandas-dev/pandas/issues/53951, instead of allowing dictionary to pass through, we should rather convert the dictionary to a plain string array?

Reply:
We can do this as a follow up, but I don't think that this is a real use case anyway

Reply:
The report in https://github.com/pandas-dev/pandas/issues/53951 is a real use case, though (and that will now create such dictionary-backed string column), AFAIU

But indeed for a different issue/PR

Reply:
isn't this also happening on main? maybe I am misunderstanding something